<!DOCTYPE html>
<html lang="de"><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta charset="utf-8">

    <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>BioASP - ASP Solutions for Systems Biology</title>

    <link href="css/bootstrap.css" rel="stylesheet">
    <style type="text/css">
      body {
        padding-top: 40px;
        padding-bottom: 40px;
        padding-left: 40px;
        padding-right: 40px;
      }
      .container {
        margin-right: auto;
        margin-left: auto;
        max-width: 900px; /* or 950px */
      }
    </style>


</head>
    
<!--  <header>

</header>-->

<!-- Inhaltsbereich: Hier gehts los! -->

    <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
         <a class="navbar-brand" href="index.html">BioASP</a>
        </div>
        <div class="collapse navbar-collapse">
          <ul class="nav navbar-nav">
<!--             <li class="active"><a href="#">BioASP</a></li> -->
            <li><a href="applications.html">Applications</a></li>
            <li><a href="publications.html">Publications</a></li>
            <li><a href="about_ASP.html">About ASP</a></li>
            <li><a href="about_us.html">About us</a></li>
            
          </ul>
        </div><!--/.nav-collapse -->
      </div>
    </div>

<body><div class="container">

    <div class="myjumbotron">
        <h1>BioASP </h1>
        <p> Answer Set Programming for Systems Biology.</p>

<!--    BioASP is a meta-package to create an powerful environment of Answer Set Programming for Systems Biology. 
    The BioASP applications implement methods for analyzing metabolic, 
    signaling and gene regulatory networks, 
    consistency checking, diagnosis, and 
    repair of biological data and models. 
    They allow for computing predictions and
    generating hypotheses about required extensions of biological models,
    as well as designing new experiments and
    finding intervention strategies to control the biological system at hand.-->
    


BioASP is a collection of open-source programs, 
 providing solutions for analyzing biological data and models with Answer Set Programming (ASP).
The BioASP applications implement methods for analyzing metabolic,
 signaling and gene regulatory networks,
 consistency checking, diagnosis, and
 repair of biological data and models. 
They allow for computing predictions and
 generating hypotheses about required extensions of biological models,
 as well as designing new experiments and
 finding intervention strategies to control the biological system at hand.
 

 
<!-- The BioASP collection is available as the <a href="http://pypi.python.org/pypi/bioasp"> bioasp</a> python package which  -->
<!--  encapsulates the ASP solutions, -->
<!--  provides functions for reading biological data formats pre-processing the input, -->
<!--  post-processing and presenting the results. -->
<!--The functionalities provided by the BioASP applications exploit technical know-how of
 modeling (biological) problems in ASP and gearing ASP solvers’ parameters to them. 
BioASP applications integrate our practical experience and offers them via easy-to-use Python functions,
 thus enabling ASP non-experts to solve biological questions with ASP.-->
 
<!--The whole BioASP collection can be installed with one <a href="http://bioasp.github.io/bioasp"> python package</a>.-->

<!--The BioASP collection is available as a [python package](http://pypi.python.org/pypi/bioasp) which encapsulates the ASP solutions,
 provides functions for reading biological data formats pre-processing the input,
 post-processing and presenting the results. -->
<!--The functionalities provided by the BioASP applications exploit technical know-how of modeling (biological) problems in ASP
 and gearing ASP solvers’ parameters to them. 
BioASP applications integrate our practical experience and offers them via easy-to-use Python functions,
 thus enabling ASP non-experts to solve biological questions with ASP.-->

<!-- <p>BioASP is powered by <a href="http://pypi.python.org/pypi/pyasp" target="_blank">  PyASP</a> and  -->
<!-- the ASP tools of <a href="http://potassco.sourceforge.net/" target="_blank"> Potassco</a>.</p> -->

<!-- Place this tag where you want the widget to render. -->
<br>
<div class="g-page" data-width="273" data-href="//plus.google.com/106623620514531759221" data-theme="dark" data-layout="landscape" data-rel="publisher"></div>

<!-- Place this tag after the last widget tag. -->
<script type="text/javascript">
  (function() {
    var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
    po.src = 'https://apis.google.com/js/platform.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
  })();
</script>
<!--<br>
Don't hesitate to ask questions or report bugs at our <a href="https://github.com/bioasp/bioasp.github.io/issues" target="_blank">issue tracker</a> 
and visit us on <a href="http://github.com/bioasp" target="_blank">github</a>.-->

</div>



<!--     <div class="span12">   -->
<!--     <h2> Here you can find the latest BioASP applications.</h2> -->

 <!--   <div class="span9">  
        <h3>Identification of Functional Gene Units</h3>
Integrating heterogeneous knowledge is necessary to elucidate the regulations in biological systems.
In particular, such an integration is widely used to identify functional units,
 that are sets of genes that can be triggered by the same external stimuli, as biological stresses, and that are linked to similar responses of the system. 
Although several models and algorithms shown great success for detecting functional units on well-known biological species,
they fail in identifying them when applied to more exotic species, such as extremophiles, that are by nature unrefined.
Shortest Genome Segments (SGS) is a new model of functional units with a predictive power that is comparable to existing methods
but overcomes this crucial limitation.
In contrary to existing methods, SGS are stable in (i) computational time and (ii) ability to predict functional units when one deteriorates the biological knowledge, 
which simulates cases that occur for exotic species.
        <br>
	<a href="http://mobyle.genouest.org/cgi-bin/Mobyle/portal.py#forms::shogen" class="btn btn-primary btn" target="_blank">WebService</a>
	<a href="http://bioasp.github.io/shogen" class="btn btn-primary btn" target="_blank">Application Website</a>
        <a href="https://pypi.python.org/pypi/shogen" class="btn btn-primary btn" target="_blank">Python Package</a>
        <br><br>
	<p><strong>Related publications:</strong></p>
	<p> <a href="http://dx.doi.org/10.1007/978-3-642-40564-8_21" target="_blank"> An ASP application in integrative biology: identification of functional gene units</a>  
        <br>  Philippe Bordron, Damien Eveillard, Alejandro Maass, Anne Siegel, Sven Thiele. </p> 
        <hr>
    </div>    -->
    

    
 <!--   <div class="span9">  
        <h3>Learning Boolean logic models for Signaling Networks         </h3>
A fundamental question in Systems Biology is the construction and training to data of mathematical models. 
Logic formalisms have become very popular to model signaling networks because their simplicity allows us to model large systems encompassing hundreds of proteins. 
An approach to train (Boolean) logic models to high-throughput phospho-proteomics data that guarantees global optimality of solutions as well as provides a complete set of solutions.

        <br>
	<a href="http://mobyle.genouest.org/cgi-bin/Mobyle/portal.py#forms::caspo-learn" class="btn btn-primary btn" target="_blank">WebService</a>
        <a href="http://bioasp.github.io/caspo" class="btn btn-primary btn" target="_blank">Application Website</a>
	<a href="http://pypi.python.org/pypi/caspo" class="btn btn-primary btn" target="_blank">Python Package</a>

        <br><br>
	<p><strong>Related publications:</strong></p>
	<p> <a href="http://dx.doi.org/10.1007/978-3-642-33636-2_20" target="_blank"> Revisiting the Training of Logic Models of Protein Signaling Networks with ASP</a>  
        <br> Santiago Videla, Carito Guziolowski, Federica Eduati, Sven Thiele, Niels Grabe, Julio Saez-Rodriguez, Anne Siegel.</p> 
        <p> <a href="http://dx.doi.org/10.1093/bioinformatics/btt393" target="_blank"> Exhaustively characterizing feasible logic models of a signaling network using Answer Set Programming</a>
        <br> Carito Guziolowski, Santiago Videla, Federica Eduati, Sven Thiele, Thomas Cokelaer, Anne Siegel, Julio Saez-Rodriguez.</p>
        <hr>
    </div>-->

<!--    <div class="span9">  
	<h3>Sign Consistency on Influence Graphs - Diagnosis, Repair, Prediction        </h3>
	Building upon a notion of consistency between biochemical/genetic reactions and high-throughput profiles of cell activity. 
	We present an approach to check the consistency of large-scale data sets, provide explanations for inconsistencies by determining minimal representations of
	conflicts. 
	In practice, this can be used to identify unreliable data or to indicate missing reactions.
	Further, we address the problem of repairing networks and corresponding yet often discrepant measurements in order to re-establish their mutual consistency and predict unobserved variations even under inconsistency.

        <br>
	<a href="http://mobyle.genouest.org/cgi-bin/Mobyle/portal.py#forms::ingranalyze" class="btn btn-primary btn" target="_blank">WebService</a>
        <a href="http://bioasp.github.io/ingranalyze" class="btn btn-primary btn" target="_blank">Application Website</a>
	<a href="http://pypi.python.org/pypi/ingranalyze" class="btn btn-primary btn" target="_blank">Python Package</a>
        
        <br><br>
	<p><strong>Related publications:</strong></p>
	<p> <a href="http://dx.doi.org/10.1007/978-3-540-89982-2_19" target="_blank"> Detecting Inconsistencies in Large Biological Networks with Answer Set Programming</a><br>
	Martin Gebser, Torsten Schaub, Sven Thiele and Philippe Veber. </p>
	<p> <a href="http://aaai.org/ocs/index.php/KR/KR2010/paper/view/1334/1660" target="_blank"> Repair and Prediction (under Inconsistency) in Large Biological Networks with Answer Set Programming </a> <br>
	Martin Gebser, Carito Guziolowski, Mihail Ivanchev, Torsten Schaub, Anne Siegel, Sven Thiele and Philippe Veber.</p>
       <hr>
    </div>-->

 <!--   <div class="span9">  
        <h3>Metabolic Network Completion         </h3>
	A qualitative approach to elaborating the bio-synthetic capacities of metabolic networks.
	Large-scale metabolic networks as well as measured data sets suffer from substantial incompleteness. 
	This approach builds upon a formal method for analyzing large-scale metabolic networks mapping its principles into Answer Set Programming.
	The approach can then be used to check whether a drafted network provides the synthesis routes to comply with the required functionality. 
	If this fails, the draft network can be completed by importing reactions from metabolic reference network stemming from other organisms until the obtained network provides the measured functionality.

        <br>
	<a href="http://mobyle.genouest.org/cgi-bin/Mobyle/portal.py#forms::meneco" class="btn btn-primary btn" target="_blank">WebService</a>
	<a href="http://bioasp.github.io/meneco" class="btn btn-primary btn" target="_blank">Application Website</a>
        <a href="http://pypi.python.org/pypi/meneco" class="btn btn-primary btn" target="_blank">Python Package</a>
        <br><br>
	<p><strong>Related publications:</strong></p>
	<p> <a href="http://dx.doi.org/10.1007/978-3-642-02846-5_27" target="_blank"> Metabolic Network Expansion with Answer Set Programming</a> <br> 
	Torsten Schaub and Sven Thiele. </p>
	<p> <a href="http://dx.doi.org/10.1007/978-3-642-40564-8_25" target="_blank"> Extending the Metabolic Network of Ectocarpus Siliculosus using Answer Set Programming</a> 
	<br> Guillaume Collet, Damien Eveillard, Martin Gebser, S. Prigent, Torsten Schaub, Anne Siegel and Sven Thiele.</p>
        <hr>
    </div>-->

<!--     <div class="span9">  
        <h3>Minimal Intervention Strategies         </h3>
	Proposing relevant perturbations to biological signaling networks is central to many problems in biology and medicine because it allows for enabling or disabling certain biological outcomes.
	In contrast to quantitative methods that permit fine-grained (kinetic) analysis,
	qualitative approaches allow for addressing large-scale networks. 
	This is accomplished by more abstract representations such as logical networks. 
	We elaborate upon such a qualitative approach aiming at the computation of minimal interventions in logical signaling networks relying on Kleene’s three-valued logic and fixpoint semantics. 
	We address this problem within answer set programming and show that it greatly outperforms previous work using dedicated algorithms.
	
        <br>
	<a href="http://mobyle.genouest.org/cgi-bin/Mobyle/portal.py#forms::caspo-control" class="btn btn-primary btn" target="_blank">WebService</a>
	<br><br>
	<p><strong>Related publications:</strong></p>
	<p> <a href="http://dx.doi.org/10.1017/S1471068413000422" target="_blank"> Minimal Intervention Strategies in Logical Signaling Networks with Answer Set Programming</a> <br> 
	Roland Kaminski, Torsten Schaub, Anne Siegel and Santiago Videla S. </p>
	<hr>
    </div>-->
<h1>
<a name="installation" class="anchor" href="#installation"><span class="octicon octicon-link"></span></a>Installation</h1>

<p>You can install all BioASP application at once from <a href="http://pypi.python.org/pypi/bioasp">pypi</a> by running:</p>

<pre><code>$ pip install bioasp
</code></pre>

<p>For details on how to use each application follow the corresponding links above. If you do not have <strong>pip</strong> installed on your system, you only need to download <a href="https://raw.github.com/pypa/pip/master/contrib/get-pip.py">get-pip.py</a>. Then run the following:</p>

<pre><code>$ python get-pip.py
</code></pre>

<p>Alternatively, if you prefer not to install <strong>pip</strong>, you can download the sources of BioASP and after unpacking run:</p>

<pre><code>$ python setup.py install
</code></pre>

<p>Note that you may need either administrator access or a <a href="http://www.virtualenv.org">virtual environment</a>.</p>

     </div> 



<!--  Ende inhalt -->

<!--  Google Tracking -->

<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-36749326-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>
<!--  Ende Google Tracking -->



  <footer>
  </footer>

    <!-- Le javascript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="js/jquery.js"></script>
    <script src="js/bootstrap-modal.js"></script>




</a></body></html>